Fixes #33632 - move react-intl to foreman-js

Ron-Lavi opened the pull request:
react-intl is being used in core and also in multiple plugins
and it makes sense to move it into foreman-js

foreman-js PR: https://github.com/theforeman/foreman-js/pull/351


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

amirfefer reviewed (commented):
what about bumping vendor-core ?
there's a small lint issue

ekohl commented on package.json:
```diff
@@ -21,19 +21,18 @@
-    "@theforeman/stories": ">= 8.7, < 8.12.1",
-    "@theforeman/test": ">= 8.7, < 8.12.1",
-    "@theforeman/vendor-dev": ">= 8.7, < 8.12.1",
+    "@theforeman/builder": ">= 8.7, < 8.15.0",
```
I think I'd keep these simple and also make them `^8.15.0`.

evgeni replied:
`^8.15.0` is not the same as `>= 8.7, < 8.15.0`, but given we bump `vendor` to `^8.15.0` using the same here sounds reasonable.

But the < 8.12.1 pin was there for a reason

Ron-Lavi replied:
Discussing with @sjha4 and @jeremylenz they say that https://github.com/theforeman/foreman-js/pull/346 got resolved so we can unpin it, let's use just `^8.15.0` then.

ekohl commented on package.json:
```diff
@@ -67,4 +66,4 @@
     "webpack-stats-plugin": "^0.1.5"
   }
-}
+}
```
Please fix the line ending.

ekohl reviewed (commented):
This will require a packaging PR as well to update the RPMs.

evgeni commented:
Is the issue with katello that @sjha4 had fixed with this release?

Ron-Lavi commented on package.json:
```diff
@@ -43,6 +41,7 @@
     "cssnano": "^4.1.10",
     "dotenv": "^5.0.0",
     "eslint": "^6.7.2",
+    "eslint-plugin-spellcheck": "0.0.17",
```
this one was moved to dependencies because the github action complained it didn't exist earlier,
let's see if returning it back to the dev dependencies still works

evgeni commented:
diff wise this looks sane to me.

Ron-Lavi commented:
packaging PR in https://github.com/theforeman/foreman-packaging/pull/7154
@evgeni @ekohl I would like to move more dependencies from plugins to foreman-js and to keep them only with dev-dependencies, so I guess there will be a few more version bumps in the following days,

also I wanted to ask, do we need to do a cleanup in foreman-packaging as we receive all of the npm modules from vendor & builder? in the past we used to add each npm package one by one to packaging.

evgeni commented:
Yeah, it's probably a good idea to walk over the packaged node modules and see which are unused by now.

Ron-Lavi commented:
@amirfefer @sharvit can you take a look please? thanks

amirfefer commented:
@laviro, LGTM but it's still blocked by packaging

ronlavi2412 committed:
Fixes #33632 - move react-intl to foreman-js

react-intl is being used in core and also in multiple plugins
and it makes sense to move it into foreman-js

foreman-js PR: https://github.com/theforeman/foreman-js/pull/351

Ron-Lavi commented:
packaging PR got merged and I also rebased this one,
ready for another round :)

Ron-Lavi commented:
A newer version is now available,
@mariaaga opened a PR in https://github.com/theforeman/foreman/pull/8863
packaging PR is merged for this PR with vendor v8.15
packaging PR to bump to v8.16 is ready to review in https://github.com/theforeman/foreman-packaging/pull/7188

EDIT:
keeping this PR open until the new packaging PR is merged.

tbrisker commented:
Included in #8863, closing
